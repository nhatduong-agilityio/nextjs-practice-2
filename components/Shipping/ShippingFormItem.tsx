import { Input, Text, Textarea, VStack } from '@chakra-ui/react'
import React, { memo } from 'react'

interface IProps {
  name: string
  type?: string
  label: string
  placeholder: string
}

const ShippingFormItem = memo(({ name, type = 'text', label, placeholder }: IProps) => (
  <VStack w='full' alignItems='flex-start' gap={['10px', '15px']} style={{ margin: 0 }} pt='15px'>
    <Text size={['16', '20']} variant='medium'>
      {label}
    </Text>
    {type !== 'textArea' ? (
      <Input
        name={name}
        type={type}
        placeholder={placeholder}
        variant='outline'
        size={['sm', '2md']}
        px='35px'
      />
    ) : (
      <Textarea
        placeholder={placeholder}
        style={{ margin: 0 }}
        background='#f9f9f9'
        fontSize={['16px', '20px']}
      />
    )}
  </VStack>
))

ShippingFormItem.displayName = 'ShippingFormItem'

export default ShippingFormItem
