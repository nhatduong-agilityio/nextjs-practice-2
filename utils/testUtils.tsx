import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
