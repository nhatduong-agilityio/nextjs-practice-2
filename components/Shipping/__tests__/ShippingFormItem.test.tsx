// Utils
import { render } from '@utils/testUtils'

import ShippingFormItem from '../ShippingFormItem'

describe('ShippingFormItem render', () => {
  const propsMock = {
    name: 'location',
    label: 'Location',
    placeholder: 'VietNam',
  }

  it('Should ShippingFormItem match Snapshot', () => {
    const { container } = render(<ShippingFormItem {...propsMock} />)

    expect(container).toMatchSnapshot()
  })
})
