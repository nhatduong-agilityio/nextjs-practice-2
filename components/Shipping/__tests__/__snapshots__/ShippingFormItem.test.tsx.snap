// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingFormItem render Should ShippingFormItem match Snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-13203fq"
    style="margin: 0px;"
  >
    <p
      class="chakra-text css-1bmwq82"
    >
      Location
    </p>
    <input
      class="chakra-input css-1nkx82e"
      name="location"
      placeholder="VietNam"
      type="text"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
